src/sensai/util/metadata.py

```diff
@@ -0,0 +1,76 @@
+from typing import Callable, Any
+
+import pandas as pd
+
+
+class DataFrameHistoryTracker:
+    """
+    A simple class for keeping track of changes in data frame metadata (like column names, and indices).
+    The idea is to call update on the result of methods that transform data frames
+
+    Example:
+
+    >>> from sensai.util.metadata import trackDFHistory, DataFrameHistoryTracker
+    >>> import pandas as pd
+
+    >>> df = pd.DataFrame({"bar": [1, 2]})
+    >>> dfHistory = DataFrameHistoryTracker(df)
+    >>> df["foo"] = [4, 5]
+    >>> df.index = ["first", "second"]
+    >>> dfHistory.update(df)
+    >>> dfHistory.indexHistory
+    [RangeIndex(start=0, stop=2, step=1), Index(['first', 'second'], dtype='object')]
+    >>> dfHistory.columnsHistory
+    [Index(['bar'], dtype='object'), Index(['bar', 'foo'], dtype='object')]
+    """
+    def __init__(self, df: pd.DataFrame):
+        self.numUpdates = -1  # after init this will be 0, so init does not count as update
+        self.columnsHistory = []
+        self.indexHistory = []
```

Comment:
Yeah, I was thinking about that too. My idea was to potentially use it in the DFTs that filter out rows but I guess it´s better to write an extension for that case instead of just saving them generally

Reply:
I made it optional in the tracker, default False. We don't track indices in the DFTs anymore